packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -778,6 +843,21 @@ private MethodBodyStatement GetPropertyInitializers(IReadOnlyList<ParameterProvi
             return propertyDeserializationStatements;
         }
 
+        private IReadOnlyList<PropertyProvider> GetAllProperties()
```

Comment:
lets see if we can use IEnumerable here to avoid the allocation of a single large list here

Reply:
well it turns out that this `IEnumerable<PropertyProvider>` will be iterate over several times later in `BuildDeserializationMethodBody`, and in `GetPropertyVariableDeclarations`, I think it should be much better if we could have the count of the properties first because we are building up an array of method body statements.
Therefore here I am changing the implementation of this method here to use `SelectMany` as the benchmark I showed you offline suggests - also I am converting it to array because we are using them several times.